x/mongo/driver/topology/connection.go

```diff
@@ -625,9 +626,6 @@ func (c *Connection) CompressWireMessage(src, dst []byte) ([]byte, error) {
 		return dst, ErrConnectionClosed
 	}
 	if c.connection.compressor == wiremessage.CompressorNoOp {
-		if len(dst) == 0 {
```

Comment:
Does the removal of this check result in an extra copy when `CompressWireMessage` is called with no compression?

Reply:
We need to guarantee `dst`, which is from the memory pool, is always returned back to the caller in all cases.